RSyntaxTextArea/src/main/java/org/fife/ui/rtextarea/SearchEngine.java

```diff
@@ -262,7 +262,7 @@ private static String getFindInText(JTextArea textArea, int start,
 			}
 		} catch (BadLocationException ble) {
 			// Never happens; findIn will be null anyway.
-			ble.printStackTrace();
+			throw new RuntimeException(ble);
```

Comment:
I'm worried this will screw up the EDT, if an uncaught exception is thrown.  At least that's my recollection.  That's why I'm currently just doing a `printStackTrace()` (which I agree with you is pretty hokey).  Is that a concern?

Reply:
Hm, I think you're right about the EDT... but if it's not re-thrown, it can't actually be caught in a test. I guess that's probably the lesser of two evils here though.

Reply:
Yeah, I'm not necessaarily happy with it but like the comment says - it's not supposed to happen :). Fortunately you caught a bug where it *does*!

RSTA follows this pattern of calling `Throwable.printStackTrace()` everywhere that e.b. `BadLocationException` is thrown by Swing API's but shouldn't happen "in real life" due to using offsets fetched from the Document, etc.  I was reluctant to rethrow checked exceptions like `BadLocationException` everywhere, since that adds complexity for client apps, and also didn't want to add a logging dependency to keep RSTA lightweight.  I'm happy folks are looking at other ways to approach this kind of thing though.

In any case, thanks for the PR!